base/array.jl

```diff
@@ -1806,15 +1806,22 @@ julia> filter(isodd, a)
 """
 filter(f, As::AbstractArray) = As[map(f, As)::AbstractArray{Bool}]
 
-function filter!(f, a::Vector)
-    insrt = 1
-    for acurr in a
-        if f(acurr)
-            a[insrt] = acurr
-            insrt += 1
+function filter!(f, a::AbstractVector)
+    if !isempty(a)
```

Comment:
That's a common style in the Julia code base for simple initial checks like this.